src/arfsdao.ts

```diff
@@ -137,27 +137,32 @@ export class ArFSDAOAnonymous extends ArFSDAOType {
 	}
 
 	async getPublicDrive(driveId: string): Promise<ArFSPublicDrive> {
+		const driveBuilder = await this.getPublicDriveMetaData(driveId);
+		return driveBuilder.build();
+	}
+
+	async getPublicDriveMetaData(driveId: string): Promise<ArFSPublicDriveBuilder> {
```

Comment:
Same comment here about a confusing API that's leaking builders without explanation.